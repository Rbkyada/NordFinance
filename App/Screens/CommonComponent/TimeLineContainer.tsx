import React from 'react';
import { Pressable, StyleSheet, View } from 'react-native';
import { BadgeBox } from '@SubComponents/BadgeBox';
import { INVESTED_TIMELINE, width } from '@Utils/Constant';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
});

interface TimeLineContainerProps {
  selectedTimeline: string;
  setSelectedTimeLine: (e: string) => void;
}

const TimeLineContainer = (props: TimeLineContainerProps) => {
  const { container } = styles;

  const { selectedTimeline, setSelectedTimeLine } = props;

  return (
    <View style={container}>
      {INVESTED_TIMELINE.map((item, index) => {
        let isSelected = selectedTimeline === item.label;
        return (
          <Pressable
            key={index}
            onPress={() => setSelectedTimeLine(item.label)}>
            <BadgeBox
              title={item.label}
              isSelect={isSelected}
              containerStyle={{
                width: width / 3.4,
                paddingVertical: 5,
              }}
              textStyle={{
                fontSize: 15,
              }}
            />
          </Pressable>
        );
      })}
    </View>
  );
};

export { TimeLineContainer };
