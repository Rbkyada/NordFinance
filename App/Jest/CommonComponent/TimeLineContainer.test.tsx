import React from 'react';
import renderer from 'react-test-renderer';
import { Pressable } from 'react-native';
import { TimeLineContainer } from '@CommonComponent/TimeLineContainer';

jest.mock('rn-range-slider', () => () => ({ run: jest.fn() }));
jest.mock('react-native-config', () => () => ({ run: jest.fn() }));
jest.mock('react-native-element-dropdown', () => () => ({ run: jest.fn() }));
jest.mock('react-native-simple-toast', () => () => ({ run: jest.fn() }));

describe('TimeLineContainer', () => {
  const INVESTED_TIMELINE = [
    { label: 'Option 1' },
    { label: 'Option 2' },
    { label: 'Option 3' },
  ];

  it('renders correctly with default props', () => {
    const selectedTimeline = 'Option 1';
    const setSelectedTimeLine = jest.fn();

    const tree = renderer
      .create(
        <TimeLineContainer
          selectedTimeline={selectedTimeline}
          setSelectedTimeLine={setSelectedTimeLine}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders the correct number of timeline buttons', () => {
    const selectedTimeline = 'Option 1';
    const setSelectedTimeLine = jest.fn();

    const component = renderer.create(
      <TimeLineContainer
        selectedTimeline={selectedTimeline}
        setSelectedTimeLine={setSelectedTimeLine}
      />,
    );

    const timelineButtons = component.root.findAllByType(Pressable);

    expect(timelineButtons.length).toBe(INVESTED_TIMELINE.length);
  });
});
