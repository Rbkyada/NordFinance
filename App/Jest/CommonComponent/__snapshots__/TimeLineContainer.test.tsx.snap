// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeLineContainer renders correctly with default props 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "alignSelf": "flex-start",
            "borderRadius": 12,
            "flexDirection": "row",
            "gap": 8,
            "justifyContent": "center",
            "overflow": "hidden",
            "paddingHorizontal": 15,
            "paddingVertical": 4,
          },
          {
            "backgroundColor": "rgba(243, 243, 243, 1)",
          },
          {
            "paddingVertical": 5,
            "width": 220.58823529411765,
          },
        ]
      }
    >
      <Text
        numberOfLines={0}
        small={true}
        style={
          [
            {
              "fontSize": 12,
            },
            {
              "color": "#1c1c1e",
            },
            {
              "fontFamily": "DMSans-Regular",
            },
            [
              {
                "color": "rgba(37, 42, 72, 0.2)",
              },
              {
                "fontSize": 15,
              },
            ],
          ]
        }
      >
        Weekly
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "alignSelf": "flex-start",
            "borderRadius": 12,
            "flexDirection": "row",
            "gap": 8,
            "justifyContent": "center",
            "overflow": "hidden",
            "paddingHorizontal": 15,
            "paddingVertical": 4,
          },
          {
            "backgroundColor": "rgba(243, 243, 243, 1)",
          },
          {
            "paddingVertical": 5,
            "width": 220.58823529411765,
          },
        ]
      }
    >
      <Text
        numberOfLines={0}
        small={true}
        style={
          [
            {
              "fontSize": 12,
            },
            {
              "color": "#1c1c1e",
            },
            {
              "fontFamily": "DMSans-Regular",
            },
            [
              {
                "color": "rgba(37, 42, 72, 0.2)",
              },
              {
                "fontSize": 15,
              },
            ],
          ]
        }
      >
        Monthly
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "alignSelf": "flex-start",
            "borderRadius": 12,
            "flexDirection": "row",
            "gap": 8,
            "justifyContent": "center",
            "overflow": "hidden",
            "paddingHorizontal": 15,
            "paddingVertical": 4,
          },
          {
            "backgroundColor": "rgba(243, 243, 243, 1)",
          },
          {
            "paddingVertical": 5,
            "width": 220.58823529411765,
          },
        ]
      }
    >
      <Text
        numberOfLines={0}
        small={true}
        style={
          [
            {
              "fontSize": 12,
            },
            {
              "color": "#1c1c1e",
            },
            {
              "fontFamily": "DMSans-Regular",
            },
            [
              {
                "color": "rgba(37, 42, 72, 0.2)",
              },
              {
                "fontSize": 15,
              },
            ],
          ]
        }
      >
        Yearly
      </Text>
    </View>
  </View>
</View>
`;
